### ISSUE TITLE
Support running application exe from a sub-directory (eg. Blazor)

### ISSUE THREAD

**Issue by witlock** [clowd/Clowd.Squirrel#121]
v3

I'm trying to update my electron.net app, but I'm getting errors such as "Cannot perform this operation in a portable app (must be installed first).". I've tried to modify the source and come up with some workarounds, but It's just problem after problem.
My guess is that this is due to the folder structure, so I'm looking for some ideas on what to do next.
The command I'm running:

`csq pack -u App -v 1.0.0 -p C:\Users\Sander\Source\Repos\App\bin\Desktop\win-unpacked -e App.exe`
In the end once I run the setup created by the command I end up with a folder structure like this (added just the parts I considered important):

`AppData\Local\App\App.exe` (The exe probably created by Squirrel that I typically run to test updating)
`AppData\Local\App\current\App.exe` (Seems to be the exe copied from Desktop\win-unpacked that you would typically use to run the electron.net app)
`AppData\Local\App\current\Squirrel.exe`
`AppData\Local\App\current\sq.version`
`AppData\Local\App\current\resources\bin\App.exe` (This is being wrapped by the exe in current\ or Desktop\win-unpacked originally to run the app as a desktop app)

basically what it looks like is that **_config** is null in squirrel.

**Comment by caesay** [comment]
For electron apps it's probably better to use v2 for now as the documentation is a bit lacking. 

In terms of the issue you're seeing, can you let me know if `AppData\Local\App\Update.exe` exists? You should not execute the Squirrel.exe which is inside of the `current` directory to perform updates.

If Update.exe is missing, can you please provide the setup logs? they should be in `AppData\Local\Squirrel.log`

**Comment by witlock** [comment]
Yes Update.exe exists and the way I attempt to update is a button that runs

`       using var mgr = new UpdateManager("domain.com/releases");
        var newVersion = await mgr.UpdateApp();
        if (newVersion != null) UpdateManager.RestartApp();`

**Comment by witlock** [comment]
v2 also doesn't work.

I'm getting
`Check 'IsInstalledApp' or 'CurrentlyInstalledVersion()' before calling this function. |Message=This operation is only valid in an installed application (Update.exe was not found).`

while Update.exe is present in the root, but it's expecting it to be in `C:\Users\Sander\AppData\Local\MyApp\app-1.0.0\resources`

once I copy it there it works.

**Comment by caesay** [comment]
Yes I see. Squirrel expects your application to be running within the app version directory (eg. `current`). It currently does not support your main executable running from a sub-directory.

**Comment by caesay** [comment]
Supported in v4
